cmd/podman/images/build.go

```diff
@@ -240,14 +240,18 @@ func buildFlagsWrapperToOptions(c *cobra.Command, contextDir string, flags *buil
 		}
 	}
 
-	pullPolicy := imagebuildah.PullNever
-	if flags.Pull {
+	pullPolicy := imagebuildah.PullIfNewer
+	if c.Flags().Changed("pull") && flags.Pull == false {
 		pullPolicy = imagebuildah.PullIfMissing
 	}
 	if flags.PullAlways {
 		pullPolicy = imagebuildah.PullAlways
 	}
 
+	if flags.PullNever {
+		pullPolicy = imagebuildah.PullNever
+	}
```

Comment:
I'm not quite sure this is quite right either 🤔 

I think the logic should be:

```go
pullPolicy :=  imagebuildah.PullIfNewer

if c.Flags().Changed("pull") {
   if flags.Pull == false {
       pullPolicy = imagebuildah.PullNever
   } else {
       pullPolicy = imagebuildah.PullAlways
  }
}
if flags.PullAlways {
    pullPolicy = imagebuildah.PullAlways
}
if flags.PullNever {
    pullPolicy = imagebuildah.PullNever
}
```

that is:
- default: pull if newer
- `--pull` => pull always (like docker)
- `--pull=true` => pull always
- `--pull=false` => pull never
- `--pull-always` => pull always
- `--pull-never` => pull never

Reply:
@asottile Maybe I need to retest, but the last time I looked at Docker it did:

```
default:  pull if missing or newer
--pull => pull if missing or newer
--pull=true => pull if missing or newer
--pull=false => pull if missing
--pull-always => pull always
--pull-never => pull never
```
I think I replicate the above with the current logic.    Personally even if different than Docker, I'd rather default, `--pull`, and `--pull=true` pull the image if missing or newer.   The `--pull=false` I could have sworn on Docker was a pull if missing too.  I'll double check that in the morning.

Thoughts from others?

Reply:
docker doesn't have `--pull-always` / `--pull-never` and docker's `--pull` acts like `--pull-always`

Reply:
So it sounds like we have one more fix needed here (`--pull` should be always instead of missing/newer) and we're good?